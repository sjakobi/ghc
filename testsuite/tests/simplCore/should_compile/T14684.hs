-- This function currently doesn't simplify using the
-- combine-identical-alternatives optimisation.
-- After the fix, Foo3 {}, Foo5 and Foo6 should be combined
-- to a single DEFAULT alternative.


module T14684 where

import GHC.List as L

data Foo = Foo1 | Foo2 | Foo3 !Int | Foo4 | Foo5 | Foo6

fun1 :: Foo -> Int
{-# NOINLINE fun1 #-}
fun1 x = case x of
               Foo1 -> 0
               Foo2 -> 1
               Foo3 {} -> 2
               Foo4 -> 1
               Foo5 -> 2
               Foo6 -> 2
