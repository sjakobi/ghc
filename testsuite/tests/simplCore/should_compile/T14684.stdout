      case w of {
        __DEFAULT -> 2#;
        Foo1 -> 0#;
        Foo2 -> 1#;
        Foo4 -> 1#
      }
